src/main/scala/nl/wbaa/gargoyle/proxy/doc/Design_considerations.md

```diff
@@ -0,0 +1,86 @@
+# Gargoyle Proxy Design considerations
+This document extends Gargoyle initial [design notes](https://hackmd.io/2DzV7iHOTwqHlqPSkbJOiQ) 
+Intention is to document all design decisions and doubts.
+
+### Proxy deployment 
+Gargoyle proxy will be placed on the same nodes that Ceph RadosGW. RGW will listen on localhost, while
+proxy process will bind to external host interface, beeing only point of entry for clients.
+Although, possible to run proxy as Docker, in such scenario proxy will be standalone process.
+
+### Stateful vs Stateless
+Proxy process should be stateless and should not maintain any data. It is "simple" data
+mover with request and policy validation.  
+
+### Scalability
+In scenario with Gargoyle proxy on RGW nodes, there will be one 2 one relation between proxy
+and RGW.
+Other options are possible, if we decide to put proxy outside RGW nodes.
+External load balancer should direct users to particular proxy.    
+
+* user sessions
+
+for now assumption is that user sessions are not taken over by other proxy processes, in 
+case of proxy crash. Client operation should be repeated then. This may be impossible in case of 
+operations like multipart upload?
+
+### Circut braker
+
+Custom mechanism should be considered to be implemented to handle situations like:
+* s3 backend is not responding or responding to slowly 
+
+### Software stack
+* Scala lang 
+* Akka Http toolkit 
+
+Http toolkit that allows transparent passing of incoming and outgoing requests.   
+
+### Proxy modes
```

Comment:
Can we make Proxy modes a level 2 header and put the NPA proxy and Pass-through proxy underneath that header? Then the structure is a bit more clear